packages/cli/src/commands/env/pull.ts

```diff
@@ -135,3 +152,79 @@ function escapeValue(value: string | undefined) {
         .replace(new RegExp('\r', 'g'), '\\r') // combine newlines (windows) into one line
     : '';
 }
+
+async function createEnvObject(
+  envPath: string,
+  output: Output
+): Promise<Dictionary<string | undefined> | undefined> {
+  try {
+    // Inspired by https://github.com/tswaters/env-file-parser/blob/master/lib/parse.js
+    let envArr = (await readFile(envPath, 'utf-8'))
+      // remove double quotes
+      .replace(/"/g, '')
+      // split on new line
+      .split(/\r?\n|\r/)
+      // filter comments
+      .filter(line => /^[^#]/.test(line))
+      // needs equal sign
+      .filter(line => /=/i.test(line));
+    return parseEnv(envArr);
+  } catch (err) {
+    output.debug(`Error parsing env file: ${err}`);
+  }
+}
+
+function findChanges(
+  oldEnv: Dictionary<string | undefined>,
+  newEnv: Dictionary<string | undefined>
+): {
+  added: string[];
+  changed: string[];
+  removed: string[];
+} {
+  let added = [];
+  let changed = [];
+  let removed = [];
+
+  for (const key of Object.keys(newEnv)) {
+    if (!oldEnv[key] && oldEnv[key] !== '') {
+      added.push(key);
+    } else if (oldEnv[key] !== newEnv[key]) {
+      changed.push(key);
+    }
+    delete oldEnv[key];
+  }
+  removed = Object.keys(oldEnv);
+
+  return {
+    added,
+    changed,
+    removed,
+  };
+}
+
+function buildDeltaString(
+  oldEnv: Dictionary<string | undefined>,
+  newEnv: Dictionary<string | undefined>
+): string {
+  let { added, changed, removed } = findChanges(oldEnv, newEnv);
+
+  let deltaString = '';
+  deltaString += chalk.green(addDeltaSection('+', added));
+  deltaString += chalk.yellow(addDeltaSection('~', changed));
+  deltaString += chalk.red(addDeltaSection('-', removed));
+
+  return deltaString;
+}
+
+function addDeltaSection(prefix: string, arr: string[]): string {
+  let deltaSection = '';
+  let ellipsis = '';
+  for (const [i, item] of arr.entries()) {
+    if (i === 2) ellipsis = '…';
```

Comment:
Should this check that there are more items as well? If there are exactly 3 items, will this render all 3 items plus an ellipses?

Can we add a test for boundary cases like this?

Reply:
Will add a test for the ellipsis case. 👍 

the intention is to render 3 items plus an ellipsis. The thinking here is that if someone made some major changes to their env variables, I wouldn't want the diff to take up too much of their screen.

Reply:
Oh, yeah, I like listing them a lot